engine/src/lib.rs

```diff
@@ -142,6 +144,20 @@ mod contract {
         io.return_output(state.owner_id.as_bytes());
     }
 
+    /// Set owner account id for this contract.
+    #[no_mangle]
+    pub extern "C" fn set_owner() {
+        let mut io = Runtime;
+        let mut state = state::get_state(&io).sdk_unwrap();
+        require_owner_only(&state, &io.predecessor_account_id());
+        let args = io.read_input().to_vec();
+        let sr: String = String::from_utf8(args).unwrap();
+        let id = AccountId::from_str(&sr).unwrap();
```

Comment:
Please avoid using the Rust unwrap. It is more efficient to use Near's host functions to abort the computation (see https://docs.near.org/sdk/rust/best-practices#stdpanic-vs-envpanic ). I would implement this as something like

```rust
let account_str = String::from_utf8(args).unwrap_or_else(|_| aurora_engine_sdk::panic_utf8(b"ERR_INVALID_STRING"));
let id: AccountId = account_str.try_into().sdk_unwrap();
```